### PR TITLE
fix: Add schema defaults for CSI

### PR DESCRIPTION
**Pull request by jimmidyson**
Also add tests for the new schema requirements.

Depends on #9

### CODE REVIEW

**Review by faiq — CHANGES_REQUESTED** [review]
Changing the name is not part of adding defaults

File: api/v1alpha1/addon_types.go [hunk 1]
```diff
@@ -191,24 +192,34 @@ func (StorageClassConfig) VariableSchema() clusterv1.VariableSchema {
...
-					XPreserveUnknownFields: true,
+					Type:        "object",
+					Description: "Parameters passed into the storage class object.",
+					AdditionalProperties: &clusterv1.JSONSchemaProps{
+						Type: "string",
+					},
+					Default: variables.MustMarshal(map[string]string{
+						"csi.storage.k8s.io/fstype": "ext4",
+						"type":                      "gp3",
+					}),
```
Comment (faiq) [inline api/v1alpha1/addon_types.go:212]: These are not the default parameters for all CSI drivers. This is specific to only AWS

**Comment by jimmidyson** [comment]
> Changing the name is not part of adding defaults

Oops! Sorry about that!

File: api/v1alpha1/addon_types.go (hunk 1 above)
Comment (jimmidyson) [reply api/v1alpha1/addon_types.go:212]: OK in that case then we can't do defaulting here and would have to do it in code.

**Review by faiq — COMMENTED** [review]
This looks good. I added some of the changes here, but I'm going to pull it down work on some changes on top of these.

**Comment by jimmidyson** [comment]
Agreed @faiq! I'll close this now but won't delete branch in case you need any of the stuff from here.

**Pull request closed without merging**